core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/jdbc/UniqueIDJDBCUserStoreManager.java

```diff
@@ -2231,6 +2241,77 @@ private void updateStringValuesToDatabase(Connection dbConnection, String sqlStm
+        boolean localConnection = false;
+        try {
+            Instant currentInstant = Instant.now();
+            if (dbConnection == null) {
+                localConnection = true;
+                dbConnection = getDBConnection();
+            }
+            prepStmt = dbConnection.prepareStatement(sqlStmt);
+            if (params != null && params.length > 0) {
+                for (int i = 0; i < params.length; i++) {
+                    Object param = params[i];
+                    if (param == null) {
+                        throw new UserStoreException("Invalid data provided");
+                    } else if (param instanceof String) {
+                        if (isStoreUserAttributeAsUnicode()) {
+                            prepStmt.setNString(i + 1, (String) param);
+                        } else {
+                            prepStmt.setString(i + 1, (String) param);
+                        }
+                    } else if (param instanceof Integer) {
+                        prepStmt.setInt(i + 1, (Integer) param);
+                    } else if (param instanceof Date) {
+                        //Convert the current date-time to UTC time with ISO Date time format.
+                        OffsetDateTime offsetDateTime = currentInstant.atOffset(ZoneOffset.UTC);
+                        LocalDateTime localDateTime = offsetDateTime.toLocalDateTime();
+                        int nanoSeconds = localDateTime.getNano();
+                        int roundedNanoSeconds = (nanoSeconds / 1000000) * 1000000;
+                        LocalDateTime formattedDateTime = localDateTime.withNano(roundedNanoSeconds);
+                        prepStmt.setTimestamp(i + 1, Timestamp.valueOf(formattedDateTime));
+                    } else if (param instanceof Boolean) {
+                        prepStmt.setBoolean(i + 1, (Boolean) param);
+                    }
+                }
+            }
+            int count = prepStmt.executeUpdate();
+
+            if (log.isDebugEnabled()) {
+                if (count == 0) {
+                    log.debug("No rows were updated");
+                }
+                log.debug("Executed query is " + sqlStmt + " and number of updated rows :: " + count);
+            }
+
+            if (localConnection) {
+                dbConnection.commit();
+            }
+        } catch (SQLException e) {
+            DatabaseUtil.rollBack(dbConnection);
+            String msg = "Error occurred while updating string values to database.";
+            if (log.isDebugEnabled()) {
+                log.debug(msg, e);
+            }
+            if (e instanceof SQLIntegrityConstraintViolationException) {
+                // Duplicate entry
+                throw new UserStoreException(msg, ERROR_CODE_DUPLICATE_WHILE_WRITING_TO_DATABASE.getCode(), e);
+            } else {
+                // Other SQL Exception
+                throw new UserStoreException(msg, e);
+            }
```

Comment:
Do not need the else block

Reply:
Addressed with https://github.com/wso2/carbon-kernel/commit/e38107689fd0a7a1dff1fb4635dca6414ccf8d41